### PR TITLE
Handle NDirect and CLR-to-COM stubs explicitly in ILStubManager::TraceManager

### LINKED ISSUE #38777: Stepping into multidimensional array of tuples in debug causes internal CLR error
### Description

Trying to step into any statement involving getting value from multidimensional array of tuples causes System.ExecutionEngineException and subsequently internal CLR error. However stepping over or checking the value through locals window does not.

### Configuration

- .Net Core 3.1 (also happens in 3.0)
- Windows
- Any CPU

### Code example

```
var array = new (int a, bool b)[2, 2];

array[0, 0] = (1, true); // Stepping in causes Internal CLR error. (0x80131506)
var k = array[0, 0]; // Stepping in causes Internal CLR error. (0x80131506)
var c = array[1, 1].a; // Stepping in causes Internal CLR error. (0x80131506)
```

### PR DESCRIPTION
**Pull request by hoyosjs**
Fallback to step over for unknown stub types.

### CODE REVIEW

**Review by jkotas — APPROVED** [review]
Thanks!

File: src/coreclr/src/vm/stubmgr.cpp [hunk 1]
```diff
@@ -1786,41 +1786,41 @@ BOOL ILStubManager::TraceManager(Thread *thread,
         LOG((LF_CORDB, LL_INFO10000, "ILSM::TraceManager: Unmanaged CALLI case 0x%x\n", target));
         trace->InitForUnmanaged(target);
     }
-    else if (pStubMD->IsStructMarshalStub())
+    else if (pStubMD->IsPInvokeStub())
```
Comment (jkoritzinsky) [inline src/coreclr/src/vm/stubmgr.cpp:1789]: Does the implementation of `IsPInvokeStub` handle other stub types like instantiating stubs correctly? It looks like its definition might not fully cover those cases.

Comment (hoyosjs) [reply src/coreclr/src/vm/stubmgr.cpp:1789]: You are actually right that some things fall through these/don't get correctly classified. I'll need to dive deeper to see why this is happening

**Comment by AaronRobinsonMSFT** [comment]
@hoyosjs Are we still planning on getting this in?

**Comment by hoyosjs** [comment]
@AaronRobinsonMSFT No. This caused even worse behavior in the debugger. I need to get back to investigating but I haven't had time lately. I'll leave the issue open and close this.

**Pull request closed without merging**